File: app/src/main/java/com/ruideraj/secretelephant/Runner.java
package com.ruideraj.secretelephant;

import android.os.Handler;

import java.util.concurrent.Executor;

public class Runner {

    private Executor mExecutor;
    private Handler mHandler;

    public Runner(Executor executor, Handler handler) {
        mExecutor = executor;
        mHandler = handler;
    }

    public void runBackground(Runnable runnable) {
        mExecutor.execute(runnable);
    }

    public void runUi(Runnable runnable) {
        mHandler.post(runnable);
    }
}

File: app/src/main/java/com/ruideraj/secretelephant/Runner.kt
package com.ruideraj.secretelephant

import android.os.Handler
import java.util.concurrent.Executor

class Runner constructor(private val executor: Executor, private val handler: Handler) {

    fun runBackground(runnable: Runnable) = executor.execute(runnable)
    fun runUi(runnable: Runnable) = handler.post(runnable)

}